packages/app-project/stores/Project.js

```diff
@@ -0,0 +1,37 @@
+import { flow, getRoot, types } from 'mobx-state-tree'
+import { get } from 'lodash'
+import asyncStates from './asyncStates'
+
+const Project = types
+  .model('Project', {
+    displayName: types.optional(types.string, ''),
+    error: types.optional(types.frozen, null),
+    id: types.optional(types.string, ''),
+    state: types.optional(types.enumeration('state', asyncStates.values), asyncStates.initialized)
+  })
+
+  .volatile(self => ({
+    client: {}
+  }))
+
+  .actions(self => ({
+    afterAttach () {
+      self.client = getRoot(self).client.projects
+    },
+
+    fetch: flow(function * fetch (slug) {
+      self.state = asyncStates.loading
+      try {
+        const project = yield self.client.get({ query: { slug }})
+          .then(response => get(response, 'projects[0]'))
+        self.displayName = project.display_name
+        self.id = project.id
+        self.state = asyncStates.success
+      } catch (error) {
+        self.error = error
```

Comment:
could we also add a `console.error(error)` here too?

Reply:
We could, but I'd like to add in a common logger / error handler so we can integrate something like Rollbar...

Reply:
I'm totally for a common logger. I just didn't realize I was missing a type error first go round since it was in the react component state and not logging to the javascript console.